### ISSUE TITLE
Server_Profile_Template_Multiconnection_Sample.ps1 - errors?

### ISSUE THREAD

**Issue by scriptengine** [HewlettPackard/POSH-HPOneView#204]
Looking at the guts of the sample: Server_Profile_Template_Multiconnection_Sample.ps1
I've not run it (as my environment does not match) but it looks to me like there as some errors in the code:

```PowerShell
$TemplateName = "Hypervisor Cluster Node Template v1"
$BL460Gen9SHT = Get-HPOVServerHardwareTypes -name "BL460c Gen9 1"

"Showing all {0} SHT without profiles assigned:" -f $BL460Gen9SHT.Name | Write-Host
Get-HPOVServer -ServerHardwareType $BL460Gen9SHT -NoProfile 

$eg                = Get-HPOVEnclosureGroup -Name "Default EG1"
$Baseline          = Get-HPOVBaseline -FileName ''
$con1              = Get-HPOVNetwork -Name "VLAN 1-A" | New-HPOVServerProfileConnection -ConnectionID 1 -Name 'VLAN 1-A Connection' -Bootable -Priority Primary
$con2              = Get-HPOVNetwork -Name "VLAN 1-B" | New-HPOVServerProfileConnection -ConnectionID 2 -Name 'VLAN 1-B Connection' -Network $Vlan1B
$con3              = Get-HPOVNetworkSet -Name 'Prod NetSet A' | New-HPOVProfileConnection -connectionId 3
$con4              = Get-HPOVNetworkSet -Name 'Prod NetSet B' | New-HPOVProfileConnection -connectionId 4
$LogicalDisk1      = New-HPOVServerProfileLogicalDisk -Name 'Disk 1' -RAID RAID1 -Bootable
$StorageController = New-HPOVServerProfileLogicalDiskController -ControllerID Embedded -Mode RAID -Initialize -LogicalDisk $LogicalDisk1

$params = @{
	Name               = $TemplateName;
	ServerHardwareType = $bl460SHT;
###
#BUG
#Should this be: ServerHardwareType = $BL460Gen9SHT
	EnclosureGroup     = $eg;
	Connections        = $con1, $con2,$con3,$con4;
	Firmware           = $true;
	Baseline           = $BaselineObject;
###
#BUG
#Should this be: Baseline           = $Baseline;
	FirmwareMode       = 'FirmwareAndSoftware'
	BootMode           = "UEFI";
	PxeBootPolicy      = "IPv4ThenIPv6";
	ManageBoot         = $True;
	BootOrder          = "HardDisk";
	LocalStorage       = $True;
	StorageController  = $StorageController;
	HideUnusedFlexnics = $True
}        
```

### Version Information
Sample from Github current branch and HPOV Powershell installation

**Comment by ChrisLynchHPE** [comment]
Yes, that is a bug, and there are some missing parameters for some of the Get Cmdlets.

**Comment by ChrisLynchHPE** [comment]
Linking with #203.

**Comment by ChrisLynchHPE** [comment]
This is fixed in [Release 3.0.1191.2858](https://github.com/HewlettPackard/POSH-HPOneView/releases/tag/v3.0.1191.2858).

Closing.

**Comment by scriptengine** [comment]
10/10 for the quick closure of this bug.
However I'm still working my way though it

In this line:
$con2              = Get-HPOVNetwork -Name "VLAN 1-B" | New-HPOVServerProfileConnection -ConnectionID 2 -Name 'VLAN 1-B Connection' -Network $Vlan1B

$Vlan1B has not been defined previously
This would throw an error if this sample was run

I'm guessing but I'd say the "-Network $Vlan1B" is not required as its an alternative to piping the network object into New-HPOVServerProfileConnection

**Comment by ChrisLynchHPE** [comment]
I'm re-releasing the update I posted last night as there was a small bug that was missed in another Cmdlet.  I also fixed the line you are referring to.  There was no need to open a new issue.

**Comment by ChrisLynchHPE** [comment]
I fixed that line and re-released the update I posted last night.  You can get it [here](https://github.com/HewlettPackard/POSH-HPOneView/releases/tag/v3.0.1191.2858_Re-release)

**Comment by scriptengine** [comment]
Further:
The sample has an old/ deprecated/ alias for New-HPOVServerProfileConnection when creating connections for Networksets. This gives the impression that a different cmdlet is require to create the network connection

$con1              = Get-HPOVNetwork -Name "VLAN 1-A" | New-**HPOVServerProfileConnection** ...
$con2              = Get-HPOVNetwork -Name "VLAN 1-B" | New-**HPOVServerProfileConnection** ...
$con3              = Get-HPOVNetworkSet -Name 'Prod NetSet A' | New-**HPOVProfileConnection** ...
$con4              = Get-HPOVNetworkSet -Name 'Prod NetSet B' | New-**HPOVProfileConnection** ...

PS> gal New-HPOVProfileConnection  | fl
DisplayName       : New-HPOVProfileConnection
CommandType       : Alias
Definition        : **New-HPOVServerProfileConnection**
ReferencedCommand : New-HPOVServerProfileConnection
ResolvedCommand   : New-HPOVServerProfileConnection

**Comment by ChrisLynchHPE** [comment]
Sorry, but that is not correct.  Yes, there is an Alias of `New-HPOVProfileConnection` that is linked to `New-HPOVServerProfileConnection`.  It is not deprecated.  I have not documented that.  It is to maintain backwards compatibility with legacy scripts, so as not to break them.

**Comment by scriptengine** [comment]
Regarding HPOVServerProfileConnection vs New-HPOVProfileConnection
My keypoint was the sample should only use the current function and not confuse the user with an alias when switching to network sets

**Comment by scriptengine** [comment]
I believe there is another bug in this sample:
The line:
$LogicalDisk1      = New-HPOVServerProfileLogicalDisk -Name 'Disk 1' -RAID RAID1 **-Bootable**
Fails
as boottable requires a boolean value and there is no default:
**PARAMETERS
    -Bootable [<Boolean>]
        Indicates if the logical drive is bootable or not. bla bla bla

        Required?                    false
        Position?                    named
        **Default value                False**
        Accept pipeline input?       false
        Accept wildcard characters?  false**

**Comment by ChrisLynchHPE** [comment]
I just looked at the documentation, and 'HPOVProfileConnection' doesn't exist in either library 2.00 or 3.00.  If you are referring to the Sample scripts, that's fair.  But do know that just because the Alias is used doesn't mean you can't view the help about the Cmdlet, or what the real Cmdlet name is.

**Comment by ChrisLynchHPE** [comment]
Not sure why the `-Bootable` parameter is a `Boolean` type.  It is supposed to be a `Switch` type.

**Comment by scriptengine** [comment]
>>Not sure why the -Bootable parameter is a Boolean type. It is supposed to be a Switch type.
I was bugging the sample

Can I leave you to bug  New-HPOVServerProfileLogicalDisk -bootable
If you think it should be a switch

**Comment by ChrisLynchHPE** [comment]
I'm only hesitant to change that due to the fact that if it is changed from `Boolean` to `Switch`, the Cmdlet call behavior would need to change, and that could break existing scripts.  `-Bootable` is not the same as `-Bootable $true`.  If the caller uses `-Bootable:$true`, then it would be the same as using `-Bootable` for a `Switch` parameter.

**Comment by ChrisLynchHPE** [comment]
Also, The Sample script is slightly wrong.  One of the Ethernet connections is set as Primary PXE boot.  You cannot combine multiple Bootable devices in a Server Profile.  So I'm going to remove the reference to `-Bootable` when creating the Logical Disk.

**Comment by scriptengine** [comment]
Additionally the line:
New-HPOVServerProfile -Name "Hyp-Clus-01" -Server $svr -ServerProfileTemplate $spt

has an error
Its requires a -assignment
The command fails without a -assignment

**Comment by ChrisLynchHPE** [comment]
This is fixed in [Release 3.0.1221.2746](https://github.com/HewlettPackard/POSH-HPOneView/releases/tag/v3.0.1221.2746).

Closing.
